File: Leetcode/Leetcode DSA course/Arrays and Strings/Sliding window Technique/5-209 Minimum size subarray sum/main.ts
function minSubArrayLen(target: number, nums: number[]): number {
  let left = 0,
    ans = Number.POSITIVE_INFINITY,
    sum = 0;
  for (let i = 0; i < nums.length; i++) {
    sum += nums[i];
    while (sum >= target) {
      ans = Math.min(ans, i - left + 1);
      sum -= nums[left++];
    }
  }
  return ans != Number.POSITIVE_INFINITY ? ans : 0;
}

File: Leetcode/Leetcode DSA course/Arrays and Strings/Sliding window Technique/5-209 Minimum size subarray sum/main.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left = 0, ans = INT_MAX, sum = 0;
        for (int right = 0; right < nums.size(); right++) {
            sum += nums[right];
            while (sum >= target) {
                ans = min(ans, right - left + 1);
                sum -= nums[left++];
            }
        }
        return (ans == INT_MAX) ? 0 : ans;
    }
};